examples/deal.II/README.MD

```diff
@@ -0,0 +1,24 @@
+An example how to write libCEED operators (BP1-BP6) within the open-source
+finite element library [deal.II](https://www.dealii.org/). As reference,
+operators are presented that use the native matrix-free infrastructure.
+
+First compile deal.II and libCEED individually. After that, compile the deal.II example:
+
+```bash
+mkdir build
+cd build
+cmake ../ -DDEAL_II_DIR=~/path/to/dealii -DCEED_DIR=~/path/to/libceed
+make
+```
+
+Run the executable to create an input file:
+
+```
+./bp > input.json
+```
+
+Modify the input file and run the executable:
+```
+mpirun -np 8 ./bp input.json
+```
```

Comment:
> What are the command line options to change the test parameters such as number of elements and FE order? (I'll read the docs and the code to help get me oriented here)

I went with a json file. Currently, number of global refinements, fe degree, and type of BP can be changed.

Reply:
Is this JSON the only way to modify the runtime arguments? I think we want a default version of the JSON in the repo for testing and then we can copy this default JSON for each test run, append the ceed backend string, and then delete the copy of the JSON once the test finishes.

Reply:
Files (and necessary care to avoid race conditions) could be avoided if the json can be read from stdin or an argument, as in `./bp "$(sed s,RESOURCE,/gpu/cuda, input.json)"`?

Reply:
Yeah, I'd prefer to avoid writing files is there is a way to use only the command line.

Reply:
I have implemented two options:
 - https://github.com/CEED/libCEED/pull/1436/commits/a09fefd3353fec5b52e44ad9514015d0372486a6 -> json can be passed via command line as suggested by @jedbrown 
 - https://github.com/CEED/libCEED/pull/1436/commits/a538308c7fe1c18af133aa14ba22f88fcfcd8662 -> all options are passed via command line; I think this would be your preference, @jeremylt?

Reply:
Any preferences? At the moment, I tend to the second option. Once agreed, I would update the readme file and squash the commits.

Reply:
Not speak for everyone, but I believe the latter (options via command line) will be the overall preferred. That's what the test suite handles most easily. 

I think even passing json via command line would require some modifications to the test suite to handle.

Reply:
Yeah, the command line options aren't that much more code/maintenance effort, and I think they'll be more familiar to libCEED users experimenting with deal.II or comparing the different implementations.

Reply:
OK. I went with the command-line arguments. What is the next step?